xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -63,6 +64,59 @@
  */
 public final class XMLUtils
 {
+    /**
+     * A simple error logging helper to suppress expected error messages.
+     * While extracting text excerpts via the {@link #extractXML} method
+     * the extraction throws expected exceptions eg. in case a text node
+     * is too long. The default ErrorListener writes these errors to the console.
+     * However as these errors are expected under normal conditions
+     * and are either handled in the {@link #extractXML} method
+     * or passed up in the chain of commands it is ok to log these
+     * at a much lower level, where users normally never see them.
+     */
+    private static class RelaxedErrorListener implements ErrorListener
+    {
+        private static final String STACK_TRACE_NOTE = "stack trace for information only";
+
+        /**
+         * Fatal errors are unexpected and are logged as warnings here.
+         * They are not really fatal to XWiki but should be handled
+         * up the command chain later, or maybe propagated to the UI level.
+         */
+        @Override
+        public void fatalError(TransformerException exception) throws TransformerException
+        {
+            LOGGER.warn("fatal error from xml transformer", exception);
+        }
+
+        /**
+         * Errors are expected and are only logged as debug.
+         * These errors happen e.g. if a text node exceeds the expected
+         * maximal length.
+         */
+        @Override
+        public void error(TransformerException exception) throws TransformerException
+        {
+            LOGGER.debug("error [{}] from xml transformer", exception.getMessage());
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace(STACK_TRACE_NOTE, exception);
+            }
+        }
+
+        /**
+         * Warnings are logged at debug level.
+         * They might be of concern for the developers but not the end users.
```

Comment:
Missing the javadoc for the parameter, that should trigger checkstyle normally

Reply:
I can trigger checkstyle intentionally when building with `-Pquality`  and adding different kind of style violations _except in the javadoc_ for the inner class. It seems not to give a damn about it as it is private, and private constructs usually do not show up in the javadoc.

I will add a `@param exception the exception to be logged` to the javadoc, but I do not know anything more specific to say about the parameter. That documentation is a bit of a waste of space IMHO.

Reply:
Indeed the class is private, that's why. No need to do anything that doesn't bring value!